src/main/kotlin/Main.kt

```diff
@@ -1,62 +1,72 @@
 import com.beust.klaxon.JsonArray
 import com.beust.klaxon.Klaxon
 import com.beust.klaxon.Parser.Companion.default
+import com.google.type.DateTime
 import facts.RelationLoader
+import parsing.EngineConfiguration
+import parsing.Sorting
 import parsing.parseEngineConfigurations
 import proofs.addAllProofs
 import tests.Test
 import tests.testgeneration.addAllTests
 import java.io.File
+import java.nio.charset.Charset
+import java.nio.file.Paths
+import java.text.SimpleDateFormat
+import java.util.Date
 import java.util.concurrent.ConcurrentLinkedQueue
 import java.util.concurrent.atomic.AtomicInteger
 import kotlin.concurrent.thread
+import kotlin.io.path.absolutePathString
 import kotlin.math.roundToInt
 import kotlin.system.measureTimeMillis
 
 fun main() {
     val time = measureTimeMillis {
-        val tests = generateTests()
-        println("Found ${tests.size} tests")
-
-        val results = executeTests(tests)
-
+        val results = executeTests()
         saveResults(results.toList())
     }
 
     println()
     println("Done in ${time / 1000} seconds")
 }
 
-private fun generateTests(): Collection<Test> {
-    val allRelations = ProofSearcher().addAllProofs().findNewRelations(RelationLoader.relations)
-    //ProofLimiter(3).limit(allRelations)
-    return TestGenerator().addAllTests().generateTests(allRelations)
-}
-
 class ResultContext(val result: TestResult,
                     val engine: String,
                     val version: String)
 
-private fun executeTests(tests: Collection<Test>): Iterable<ResultContext> {
+private fun executeTests(): Iterable<ResultContext> {
     val engines = parseEngineConfigurations()
     val fullResults = ArrayList<ResultContext>()
+    val allTests = generateTests()
+    println("Found ${allTests.size} tests")
 
     for (engine in engines) {
+        val tests = sortTests(engine, allTests)
+        if (engine.omitTests == true) { //Omitting tests and/or saving them
+            val path = engine.testsSavePath ?: "./${engine.name}_tests" //TODO : Default file
+            saveTests(engine.name, path, tests)
+            println("Executing the tests for ${engine.name} have been omitted")
+            println()
+            continue
```

Comment:
I feel like this "continue" might be misplaced. Because if we want to "omitTests" then why are we in the function "executeTests"?